backend/app/core/rag/retriever.py

```
@@ -0,0 +1,193 @@
+import re
+import numpy as np
+from typing import List, Dict, Any
+from sentence_transformers import CrossEncoder
+from app.core.config import settings
+from .embedder import EmbeddingService
+from .graph_builder import _chunk_node_id, KnowledgeGraphBuilder
+import networkx as nx
+
+class HybridRetriever:
+    #Гибридный поиск: FAISS + BM25 + RRF + CrossEncoder + граф знаний
+    def __init__(self, embedder: EmbeddingService, top_k: int = None,
```

Ruff (0.15.12)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)